pkgs/applications/virtualization/arion/default.nix

```diff
@@ -0,0 +1,26 @@
+{ pkgs
+, haskellPackages ? pkgs.haskellPackages
+}:
```

Comment:
Instead of taking `pkgs` as an argument here, could you only declare the things you need from `pkgs`?  That would make it more natural to use as a `callPackage`-style package.

I'm specifically thinking of things like `haskellPackages` (which you do declare), `haskell` (for `haskell.lib`, or even just the specific things you use from `haskell.lib`).